lib/purchases_flutter.dart

```diff
@@ -32,17 +32,19 @@ class Purchases {
   /// [observerMode] An optional boolean. Set this to TRUE if you have your own
   /// IAP implementation and want to use only RevenueCat's backend.
   /// Default is FALSE.
+  /// 
+  /// [userDefaultsSuiteName] iOS-only, will be ignored for Android. 
+  /// Set this if you would like the RevenueCat SDK to store its preferences in a different
+  /// NSUserDefaults suite, otherwise it will use standardUserDefaults.
+  /// Default is null, which will make the SDK use standardUserDefaults.
   static Future<void> setup(String apiKey,
-      {String appUserId, bool observerMode = false}) async {
-    if (appUserId == null) {
-      return await _channel.invokeMethod('setupPurchases', {'apiKey': apiKey});
-    } else {
```

Comment:
removed this if else block, and replaced it with actually checking if the value is NSNull and translating to nil.

Reply:
note that there was a bug here - if you passed in `observerMode` but not `appUserId`, `observerMode` is ignored

Reply:
funny there was a similar error in react native

Reply:
Just wanted to double check that removing this if else doesn't break Android

Reply:
yep, it works correctly in Android